compiler/data/class-data.h

```diff
@@ -23,6 +23,12 @@ struct FFIClassDataMixin;
 struct FFIScopeDataMixin;
 namespace kphp_json { class KphpJsonTagList; }
 
+enum class SubtreeImmutableType {
+  immutable,
+  not_immutable,
```

Comment:
I would change at `mutable`, but up to you.

Reply:
mutable is a key word